### PR TITLE
Set get item data

### PR DESCRIPTION
**Pull request by ambroseus**
Added helpful action/selector to have ability to set/get Item specific data for ex. **totalPages, currentPage, importStatus, validateStatus** etc.
`setTotalPages: (itemType, totalPages) =>
    DckActionCreators.setItemData(itemType, "totalPages", totalPages)`
`const getTotalPages = (state, itemType) =>
DckSelectors.getItemData(state, itemType, "totalPages");`

### CODE REVIEW

**Comment by volodymyrrudyi** [comment]
@ambroseus  Please add a description to the PR

**Comment by ambroseus** [comment]
> @ambroseus Please add a description to the PR

added

**Comment by ambroseus** [comment]
checked with real project - works as expected

File: packages/dck-redux/src/actions/index.ts [hunk 1]
```diff
@@ -157,8 +163,16 @@ export class DckActionCreators {
    * @param verificationCode verification code
    * @param password new user password
    */
-  public static confirmForgotPassword(email: string, verificationCode: string, password: string) {
-    return AccountActionCreators.confirmForgotPassword(email, verificationCode, password);
+  public static confirmForgotPassword(
```
Comment (volodymyrrudyi) [inline packages/dck-redux/src/actions/index.ts]: Reformatting of unrelated code

File: packages/dck-redux/src/actions/index.ts [hunk 2]
```diff
@@ -314,7 +339,11 @@ export class DckActionCreators {
    * @param filter filter name
    * @param values filter values
    */
-  public static setItemSearchFilters(itemType: string, filter: any, values: any) {
+  public static setItemSearchFilters(
```
Comment (volodymyrrudyi) [inline packages/dck-redux/src/actions/index.ts]: unrelated code reformatting. Please submit a separate PR with formatting changes.

File: packages/dck-redux/src/actions/index.ts [hunk 3]
```diff
@@ -363,7 +392,12 @@ export class DckActionCreators {
     const object: any = {};
     if (keys && keys.length) {
       keys.forEach((key: string) => {
-        if (key && key !== "getBindObject" && DckActions[key] && typeof DckActions[key] === "function") {
+        if (
```
Comment (volodymyrrudyi) [inline packages/dck-redux/src/actions/index.ts]: unrelated reformatting

File: packages/dck-redux/src/actions/items.ts [hunk 4]
```diff
@@ -224,7 +240,11 @@ export function removeItemSearchFilter(itemType: string, filter: any) {
  * @param values filter values
  * @hidden
  */
-export function setItemSearchFilters(itemType: string, filter: any, values: any) {
+export function setItemSearchFilters(
```
Comment (volodymyrrudyi) [inline packages/dck-redux/src/actions/items.ts]: same

File: packages/dck-redux/src/actions/types.ts [hunk 5]
```diff
@@ -53,7 +53,8 @@ export class DckActionTypes {
   /**
    * Ensure changed user password is set action type.
    */
-  public static ENSURE_CHANGE_USER_PASSWORD_IS_SET = "ENSURE_CHANGE_USER_PASSWORD_IS_SET";
+  public static ENSURE_CHANGE_USER_PASSWORD_IS_SET =
```
Comment (volodymyrrudyi) [inline packages/dck-redux/src/actions/types.ts]: same

File: packages/dck-redux/src/reducers/items.ts [hunk 6]
```diff
@@ -67,10 +70,18 @@ export const createItemsReducer = (itemTypes: string[]) => {
         const previousActive = state.getIn([action.itemType, "active"]);
         const previousTerm = state.getIn([action.itemType, "term"]);
         const previousFilters = state.getIn([action.itemType, "filters"]);
-        const previousSortingOptions = state.getIn([action.itemType, "sortingOptions"]);
-        const newItems = (action.data && action.id) 
-                          ? state.getIn([action.itemType, "items"]).toJS().filter((item: any) => item.id !== action.id).concat([{...action.data, id: action.id}])
-                          : state.getIn([action.itemType, "items"]).toJS();
+        const previousSortingOptions = state.getIn([
```
Comment (volodymyrrudyi) [inline packages/dck-redux/src/reducers/items.ts]: unrelated reformatting

File: packages/dck-redux/src/reducers/items.ts [hunk 7]
```diff
@@ -83,20 +94,27 @@ export const createItemsReducer = (itemTypes: string[]) => {
...
       [DckActionTypes.SET_ITEM_SEARCH_TERM](state: any, action: any) {
         return state.setIn([action.itemType, "term"], action.term);
       },
       [DckActionTypes.ADD_ITEM_SEARCH_FILTER](state: any, action: any) {
-        return state.updateIn([action.itemType, "filters", action.filter], List(), (list: any) => {
-          if (!list.includes(action.value)) {
-            return list.push(action.value);
-          } else {
-            return list;
+        return state.updateIn(
```
Comment (volodymyrrudyi) [inline packages/dck-redux/src/reducers/items.ts]: same here

File: packages/dck-redux/tslint.json [hunk 8]
```diff
@@ -5,7 +5,8 @@
   "rules": {
     "ordered-imports": false,
     "object-literal-sort-keys": false,
-    "interface-over-type-literal": false
+    "interface-over-type-literal": false,
+    "no-console": false
```
Comment (volodymyrrudyi) [inline packages/dck-redux/tslint.json]: please don't add this ignore rule. We are providing a lib, we should not be using console

**Comment by ambroseus** [comment]
created new PR according to review comments. closing
